server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessor.java

```diff
@@ -41,4 +43,24 @@
       @NonNull String code,
       @NonNull Deque<CopybookUsage> copybookStack,
       @NonNull CopybookProcessingMode copybookProcessingMode);
+
+  /**
+   * Build extended document using its COPY statements, excluding non-processable statements,
+   * applying related semantic analysis
+   *
+   * @param uri - the URI of the processing document
+   * @param code - COBOL program text to analyse
+   * @param copybookStack - stack, representing the hierarchy of nesting copybooks
+   * @param copybookProcessingMode - reflects the status of the copybook support (ENABLED|DISABLED)
+   * @param positionCorrectionMap - contains mapping between the modified new line in document as
+   *     per preprocessing to the actual {@link CobolLine} in the document
+   * @return extended document with copybooks and related errors
+   */
+  @NonNull
+  ResultWithErrors<ExtendedDocument> buildExtendedDocument(
```

Comment:
Are you using both of those methods?

Reply:
yes, we may get rid of one which calls the other one.